Source/OCMockTests/OCMockObjectCleanupTests.m

```diff
@@ -0,0 +1,95 @@
+/*
+ *  Copyright (c) 2015-2020 Erik Doernenburg and contributors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use these files except in compliance with the License. You may obtain
+ *  a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+
+#import <XCTest/XCTest.h>
+#import <OCMock/OCMock.h>
+
+#pragma mark   Helper classes
+
+// Tests for mocks being stopped by the XCTestObserver that we register in OCMockObject.
+@interface OCMockObjectCleanupTests : XCTestCase
+@end
+
+@implementation OCMockObjectCleanupTests
+
+
+static id suiteMock;
+static id caseMock;
+
++ (void)setUp
+{
+    suiteMock = [OCMockObject mockForClass:[NSString class]];
+    OCMStub([suiteMock intValue]).andReturn(42);
+    caseMock = nil;
+}
+
+#pragma mark   Tests suite mocks survive across test cases
+
+- (void)testSuiteMockWorksHere
+{
+    // Verify that a testSuite Mock made in +setUp doesn't get cleaned up until test suite is done.
+    // By verifying in two test cases we know this is true (See testSuiteMockWorksAndHere).
+    XCTAssertEqual([suiteMock intValue], 42);
+}
+
+- (void)testSuiteMockWorksAndHere
+{
+    // Verify that a testSuite Mock made in +setUp doesn't get cleaned up until test suite is done.
+    // By verifying in two test cases we know this is true (See testSuiteMockWorksHere).
+    XCTAssertEqual([suiteMock intValue], 42);
+}
```

Comment:
I see what you're doing with these tests, but you these tests only demonstrate that the cleanup doesn't happen under incorrect circumstances. There's no test to show that the cleanup actually happens. Or, in other words, these tests would pass without the functionality added.